task_1.py

```diff
@@ -0,0 +1,198 @@
+        self.num_pages = num_pages
+        self.headers = {
+            "accept": "application/json",
+            "Authorization": "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIzMTI3NGFmYTRlNTUyMjRjYzRlN2Q0NmNlMTNkOTZjOSIsInN1YiI6IjVkNmZhMWZmNzdjMDFmMDAxMDU5NzQ4OSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.lbpgyXlOXwrbY0mUmP-zQpNAMCw_h-oaudAJB6Cn5c8"
+        }
+        self.movies = []
+        self.fetch_data()
+
+    def fetch_data(self):
+        for page_num in range(1, self.num_pages + 1):
+            url = f"https://api.themoviedb.org/3/discover/movie?include_adult=false&include_video=false&sort_by=popularity.desc&page=" + str(
+                page_num)
+            response = requests.get(url, headers=self.headers)
+            if response.status_code == 200:
+                movies = response.json()['results']
+                for movie in movies:
+                    genres = self.get_genre_names(movie['genre_ids'])
+                    movie['genre_names'] = genres
+                self.movies.extend(movies)
+            else:
+                print(f"Failed to fetch data from page {page_num}.Status code: {response.status_code}")
+
+    def get_genre_names(self, ids):
+        url = 'https://api.themoviedb.org/3/genre/movie/list?language=en'
+        response = requests.get(url, headers=self.headers)
+        genre_names = []
+        if response.status_code == 200:
+            genre_data = response.json()
+            genres = genre_data.get('genres', [])
+            for id in ids:
+                for genre in genres:
+                    if genre['id'] == id:
+                        genre_names.append(genre['name'].lower())
+                        break
+        return genre_names
+
+    def get_movies_by_indexes(self, start, end, step):
+        if step == 0:
+            raise ValueError("Step cannot be zero")
+        return self.movies[start:end+1:step]
+
+    def get_all_data(self):
+        return self.movies
+
+    def get_most_popular_title(self):
+        maximum_popularity = float('-inf')
+        most_popular_movie_title = None
+        for movie_data in self.movies:
+            if movie_data['popularity'] > maximum_popularity:
+                maximum_popularity = movie_data['popularity']
+                most_popular_movie_title = movie_data['title']
+        if most_popular_movie_title is not None:
+            return most_popular_movie_title
+        else:
+            return 'Error'
+
+    def get_titles_by_keyword(self, search_keyword):
+        titles = [movie['title'] for movie in self.movies if search_keyword.lower() in movie['title'].lower()]
+        if not titles:
+            return ["No movies found with the specified keyword."]
+        return titles
+
+    def get_genres(self):
+        genres = set()
+        for movie in self.movies:
+            genres.update(movie['genre_names'])
+        return genres
+
+    def delete_movies_with_genre(self, genre_id_or_name):
```

Comment:
avoid using parameters for a few purposes. better to create two separate parameters